templates/makefile-c++/README.md

```diff
@@ -0,0 +1,38 @@
+# C++ Makefile Project Template
+
+## About this Project
+This is a basic makefile template designed for C++. 
+
+### Supported Platforms 
+The contents of the makefile are generalized to support compilation to target either Windows or Linux out of the box. You may need to include platform specific headers as you go, which may restrict your project to a single platform. Take some time to consider what platforms you plan to target, or if cross-compatibility is required.
+
+__*Note: This project has been only tested for Windows using the Git Bash command line interface. Using PowerShell, cmd.exe, or other, non-unix shells may result in errors using make and g++.*__
```

Comment:
I don't think that the `gcc` toolchain or `make` ships with Git Bash by default (From testing on my Windows machine). Perhaps include a note on downloading the gcc toolchains on Windows? Looks as though you may have it from a [Cygwin install](https://gcc.gnu.org/install/binaries.html)?

Reply:
> I don't think that the `gcc` toolchain or `make` ships with Git Bash by default (From testing on my Windows machine). Perhaps include a note on downloading the gcc toolchains on Windows? Looks as though you may have it from a [Cygwin install](https://gcc.gnu.org/install/binaries.html)?

Testing this on my laptop looks like you are correct - I will see about adding a section about installing `make` for the Windows Toolchain. 

Another idea - a batch script that installs and sets-up a proper Windows environment (git bash, vs-code, gcc/make toolchain, etc.) all in one go to cut down on incosistent setups?

Reply:
A script could work. Although if we want to pre-package the development environment, I would favor a docker container. Since VS-Code integrates with that very nicely and we ensure a perfectly identical build environment across the board.

Reply:
That is true - something to think about. I added the warning about missing binaries in Windows, should be good for now. I think we should consider containerizing our environments if the end-user complexity is minimal.